stdlib/2and3/ssl.pyi

```diff
@@ -172,11 +173,17 @@ if sys.version_info < (3,) or sys.version_info >= (3, 4):
     ALERT_DESCRIPTION_UNSUPPORTED_EXTENSION: int
     ALERT_DESCRIPTION_USER_CANCELLED: int
 
-if sys.version_info < (3,) or sys.version_info >= (3, 4):
-    _PurposeType = NamedTuple('_PurposeType', [('nid', int), ('shortname', str), ('longname', str), ('oid', str)])
-    class Purpose:
-        SERVER_AUTH: _PurposeType
-        CLIENT_AUTH: _PurposeType
+if sys.version_info < (3,):
+    class _ASN1Object(NamedTuple('_ASN1Object', [('nid', int), ('shortname', str), ('longname', str), ('oid', str)])): ...
+    class Purpose(_ASN1Object):
+        SERVER_AUTH: Purpose
+        CLIENT_AUTH: Purpose
+if sys.version_info >= (3, 4):
+    class _ASN1Object(NamedTuple('_ASN1Object', [('nid', int), ('shortname', str), ('longname', str), ('oid', str)])): ...
+    class Purpose(_ASN1Object, enum.Enum):
+        SERVER_AUTH: Purpose
+        CLIENT_AUTH: Purpose
+        value: _ASN1Object
```

Comment:
`value` should not be needed. Type checkers should special-case enums.

Reply:
Emm, it seems my local mypy version is old.
```Python
from enum import Enum


class Color(Enum):
    red = 0
    blue = 1


def test(a: str) -> str:
    return a


test(Color.red.value)
```
mypy 0.701 could not find the type problem in above code(but 0.710 can), so I add the extra code to force check.